library/core/src/fmt/num_buffer.rs

```diff
@@ -0,0 +1,59 @@
+use crate::mem::MaybeUninit;
+use crate::slice::SliceIndex;
+
+/// 40 is chosen as the buffer length, as it is equal
+/// to that required to accommodate i128::MIN, which has the largest
+/// decimal string representation
+/// (39 decimal digits + 1 for negative sign).
+const BUF_SIZE: usize = 40;
+
+/// A minimal buffer implementation containing elements of type
+/// `MaybeUninit<u8>`.
+#[unstable(feature = "int_format_into", issue = "138215")]
+#[derive(Debug)]
+pub struct NumBuffer {
+    /// An array of elements of type `MaybeUninit<u8>`.
+    pub contents: [MaybeUninit<u8>; BUF_SIZE],
```

Comment:
I don't see a reason why this should be `pub`. Perhaps `pub(crate)` if a different module inside core (core::fmt::num) needs to access it.